android/app/src/main/java/app/covidshield/receiver/worker/ExposureCheckHeadlessTask.java

```diff
@@ -99,9 +99,14 @@ public void run() {
     }
 
     private void invokeStartTask(ReactContext reactContext, final HeadlessJsTaskConfig taskConfig) {
-        if (reactContext.getLifecycleState() == LifecycleState.RESUMED) {
-            MetricsService.publishDebugMetric(3.9, reactContext);
-            return;
+
+        try {
+            if (reactContext.getLifecycleState() == LifecycleState.RESUMED) {
+                MetricsService.publishDebugMetric(3.9, reactContext);
+                return;
+            }
+        } catch (Exception exception) {
+            MetricsService.publishDebugMetric(109, reactContext, exception.getMessage());
```

Comment:
I think we should return here, something has gone wrong and I'm not sure if we can really continue?